dev/TabView/InteractionTests/TabViewTests.cs

```diff
@@ -0,0 +1,237 @@
+            }
+        }
+
+        [TestMethod]
+        public void IsCloseableTest()
+        {
+            using (var setup = new TestSetupHelper("TabView Tests"))
+            {
+                UIObject firstTab = FindElement.ByName("FirstTab");
+                Button closeButton = FindCloseButton(firstTab);
+                Verify.IsNotNull(closeButton);
+
+                Log.Comment("Setting IsCloseable=false on the first tab.");
+                CheckBox isCloseableCheckBox = FindElement.ByName<CheckBox>("IsCloseableCheckBox");
+                isCloseableCheckBox.Uncheck();
+                Wait.ForIdle();
+
+                ElementCache.Refresh();
+                closeButton = FindCloseButton(firstTab);
+                Verify.IsNull(closeButton);
+
+                Log.Comment("Setting IsCloseable=true on the first tab.");
+                isCloseableCheckBox.Check();
+                Wait.ForIdle();
+
+                ElementCache.Refresh();
+                closeButton = FindCloseButton(firstTab);
+                Verify.IsNotNull(closeButton);
+
+                Log.Comment("Setting CanCloseTabs=false on the TabView.");
+                CheckBox canCloseCheckBox = FindElement.ByName<CheckBox>("CanCloseCheckBox");
+                canCloseCheckBox.Uncheck();
+                Wait.ForIdle();
+
+                ElementCache.Refresh();
+                closeButton = FindCloseButton(firstTab);
+                Verify.IsNull(closeButton);
+            }
+        }
+
+        [TestMethod]
+        public void CancelTabClosingTest()
+        {
+            using (var setup = new TestSetupHelper("TabView Tests"))
+            {
+                UIObject firstTab = FindElement.ByName("FirstTab");
+                Button closeButton = FindCloseButton(firstTab);
+                Verify.IsNotNull(closeButton);
+
+                CheckBox cancelCloseCheckBox = FindElement.ByName<CheckBox>("CancelCloseCheckBox");
+                cancelCloseCheckBox.Check();
+                Wait.ForIdle();
+
+                Log.Comment("Clicking close button should not close tab if app returns cancel = true.");
+                closeButton.Invoke();
+                Wait.ForIdle();
+
+                ElementCache.Refresh();
+                firstTab = TryFindElement.ByName("FirstTab");
+                Verify.IsNotNull(firstTab);
+
+                cancelCloseCheckBox.Uncheck();
+                Wait.ForIdle();
+
+                Log.Comment("Clicking close button should close tab if app doesn't handle TabClosing event.");
+                closeButton.Invoke();
+                Wait.ForIdle();
+
+                ElementCache.Refresh();
+                firstTab = TryFindElement.ByName("FirstTab");
+                Verify.IsNull(firstTab);
+            }
+        }
+
+        Button FindCloseButton(UIObject tabItem)
+        {
+            foreach (UIObject elem in tabItem.Children)
+            {
+                if (elem.ClassName.Equals("Button"))
```

Comment:
There's no other button inside the tab. Unfortunately, as there are multiple tabs, FindElement doesn't work because I can't tell it to find the specific one I'm looking for, so I have to look specifically inside the tabItem's children.